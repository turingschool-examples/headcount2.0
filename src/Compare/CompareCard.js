import React from 'react';
import './ComparedCard.css';
import PropTypes from 'prop-types';

const CompareCard = ({compareInfo}) => {
  const newCompareInfo = Object.keys(compareInfo);
  return (
    <div className="card clicked compare-card">
      <h3>
        {`${newCompareInfo[0]}: ${compareInfo[newCompareInfo[0]]}`}
      </h3>
      <h3>
        {`<------ ${compareInfo.compared} ------>`}
      </h3>
      <h3>
        {`${newCompareInfo[1]}: ${compareInfo[newCompareInfo[1]]}`}
      </h3>
    </div>
  );
};

CompareCard.propTypes = {
  compareInfo: PropTypes.object
};

export default CompareCard;