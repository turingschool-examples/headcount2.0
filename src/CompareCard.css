@import url('https://fonts.googleapis.com/css?family=Roboto');

.compare-card {
  border: 5px solid #337AB7;
  border-radius: 10%;
  width: 300px;
  margin: 20px;  
  background-color: #fff;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.location {
  margin-bottom: 0;
}

.location-num {
  color:#337AB7;
}

.compare-num {
  color: #F96854;
  font-size: 35px;
}

span {
  color: #000;
}
