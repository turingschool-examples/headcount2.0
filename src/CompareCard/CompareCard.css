.CompareCard {
	background-color: #E9EBEE;
	display: flex;
	justify-content: center;
  padding: 0 0 1.5rem 0;
  margin: auto;
  width: 50%;
}

.average-card {
	align-items: center;
	background-color: #FFF;
	border-radius: 10px;
	box-shadow: 0px 0px 1px 2px #F5F6F7;
	cursor: pointer;
	display: flex;
	flex-direction: column;
	height: 26rem;
	justify-content: space-around;
	margin: 0 2rem 0.5rem;
	padding: 0 0 1rem;
	text-align: center;
	width: 15rem;
}

.location {
  font-family: 'Arvo', serif;
  margin: 1rem 2rem 0;
}

.average {
	background-color: #F5F6F7;
	border-radius: 10px;
	color: #01411b;
	font-family: 'Open Sans', sans-serif;
	font-weight: 300;
	margin: 0.2rem auto;
	padding: 0 0.75rem;
}

.compared {
	color: #087ff5;
	font-size: 1.5rem;
}

.label-average, .label-location, .label-compared {
	color: #026332;
	font-family: 'Open Sans', sans-serif;
	font-size: 1rem;
}