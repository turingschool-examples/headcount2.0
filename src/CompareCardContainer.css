.compare-cards {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  align-items: center;
}

.selected {
  border: 5px solid #337AB7;
}