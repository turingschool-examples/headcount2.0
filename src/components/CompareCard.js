import React from 'react';


const CompareCard = (props) => {
  const displayCompareCard = Object.keys(props).map((key, index) => {
    return (
      <p key={index} className='comparedStats'>{key}: {props[key]}</p>
    )
  })
  return (
    <article className='card compareCard active'>
      {displayCompareCard}
    </article>
  )
}

export default CompareCard;
