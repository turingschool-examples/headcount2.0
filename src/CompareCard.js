import React from 'react';
import PropTypes from 'prop-types';
import './CompareCard.css';

const CompareCard = (props) => {
  let keys = Object.keys(props.cardInfo)
  let values = Object.values(props.cardInfo)
  let leftArrow = '<----'
  let rightArrow = '---->'


  return (
    <div className="compare-card">
      <h1 className="location">{keys[0]}:</h1>
      <h2 className="location-num">{values[0]}</h2>
      <h1 className="compare-num"><span>{leftArrow}</span> {values[2]} <span>{rightArrow}</span></h1>
      <h1 className="location">{keys[1]}:</h1>
      <h2 className="location-num">{values[1]}</h2>
    </div>

  )

}


export default CompareCard;