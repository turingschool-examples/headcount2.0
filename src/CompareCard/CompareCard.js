import React from 'react';
import './CompareCard.css';
import DistrictCard from '../DistrictCard/DistrictCard.js';
import PropTypes from 'prop-types';

const CompareCard = ({ districtsBeingCompared, districtAverages }) => {
	if (districtsBeingCompared) {
	const districtCards = districtsBeingCompared.map((district) => {
		return <DistrictCard 
							key={district.location}
							schoolName={district.location}
							{...district}
						/>
	})

	if (districtsBeingCompared.length === 2 && districtAverages) {
		console.log(districtAverages)
		return (
			<div className='CompareCard'>
				{ districtCards[0] }
				<div className='average-card'>
					<h3 className='location'>{districtsBeingCompared[0].location}</h3>
					<p className='label-average'>Average:</p>
					<p className='average'>{districtAverages[districtsBeingCompared[0].location]}</p>
					<p className='label-compared'>Compared:</p>
					<h3 className='compared'>{districtAverages.compared}</h3>
					<h3 className='location'>{districtsBeingCompared[1].location}</h3>
					<p className='label-average'>Average:</p>
					<p className='average'>{districtAverages[districtsBeingCompared[1].location]}</p>
				</div>
				{ districtCards[1] }
			</div>
		)
	} else {
			return (
				<div className='CompareCard'>
					{ districtCards }
				</div>
			)
		}

	} 
}

CompareCard.propTypes = {
	districtsBeingCompared: PropTypes.array
}

export default CompareCard;