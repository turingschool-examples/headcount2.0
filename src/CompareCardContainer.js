import React from 'react';
import Card from './Card.js';
import CompareCard from './CompareCard.js';
import './CompareCardContainer.css';

const CompareCardContainer = (props) => {
  let card1;
  let card2;
  let middleCard;

  if(props.appState.compareCard1 !== null) {
    card1= <Card cardInfo={props.appState.compareCard1} />
  }
  if(props.appState.compareCard2 !== null) {
    card2= <Card cardInfo={props.appState.compareCard2} />
  }
  if(props.appState.compareCard1 !== null && props.appState.compareCard2 !== null) {
    let object = props.compareCards(props.appState.compareCard1.location, props.appState.compareCard2.location)
    console.log('middlecard', object)
    middleCard= <CompareCard cardInfo={object} />
  }



 

  return (
    <div className="compare-cards">{card1}{middleCard}{card2}</div>

  )
}

export default CompareCardContainer;