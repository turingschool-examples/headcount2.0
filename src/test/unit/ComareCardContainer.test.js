import React from 'react';
import Data from '../../data/kindergartners_in_full_day_program.js';
import CompareCardContainer from '../../CompareCardContainer.js';
import CompareCard from '../../CompareCard.js';
import {shallow, mount} from 'enzyme';

describe('CompareCardContainer', () => {
  it('should match the snapshot with all data passed in correctly', () => {
    const card1 = {
      location: "ADAMS COUNTY 14",
      selected: false,
      stats:
      {2004: 0.228,
      2005: 0.3,
      2006: 0.293,
      2007: 0.306,
      2008: 0.673,
      2009: 1,
      2010: 1,
      2011: 1,
      2012: 1,
      2013: 0.998,
      2014: 1
      }
    }

    const card2 = {
      location: "ADAMS COUNTY 14",
      selected: false,
      stats:
      {2004: 0.228,
      2005: 0.3,
      2006: 0.293,
      2007: 0.306,
      2008: 0.673,
      2009: 1,
      2010: 1,
      2011: 1,
      2012: 1,
      2013: 0.998,
      2014: 1
      }
    }

    const state = {
      compareCard1: card1,
      compareCard2: card2
    }

    const wrapper = shallow(
      <CompareCardContainer appState={state} />
    );
    
    expect(wrapper).toMatchSnapshot()
  })


})