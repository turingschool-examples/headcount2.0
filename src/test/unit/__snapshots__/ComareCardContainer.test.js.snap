// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompareCardContainer should match the snapshot with all data passed in correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    data={
        Object {
            "CompareCard2": Object {
              "location": "ADAMS COUNTY 14",
              "stats": Object {
                "2004": 0.228,
                "2005": 0.3,
                "2006": 0.293,
                "2007": 0.306,
                "2008": 0.673,
                "2009": 1,
                "2010": 1,
                "2011": 1,
                "2012": 1,
                "2013": 0.998,
                "2014": 1,
              },
            },
            "compareCard1": Object {
              "location": "ADAMS COUNTY 14",
              "stats": Object {
                "2004": 0.228,
                "2005": 0.3,
                "2006": 0.293,
                "2007": 0.306,
                "2008": 0.673,
                "2009": 1,
                "2010": 1,
                "2011": 1,
                "2012": 1,
                "2013": 0.998,
                "2014": 1,
              },
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          cardInfo={
                    Object {
                              "location": "ADAMS COUNTY 14",
                              "stats": Object {
                                "2004": 0.228,
                                "2005": 0.3,
                                "2006": 0.293,
                                "2007": 0.306,
                                "2008": 0.673,
                                "2009": 1,
                                "2010": 1,
                                "2011": 1,
                                "2012": 1,
                                "2013": 0.998,
                                "2014": 1,
                              },
                            }
          }
          compareCard1={undefined}
          compareCard2={undefined}
          displaySelected={undefined}
/>,
        <Card
          cardInfo={
                    Object {
                              "location": "ADAMS COUNTY 14",
                              "stats": Object {
                                "2004": 0.228,
                                "2005": 0.3,
                                "2006": 0.293,
                                "2007": 0.306,
                                "2008": 0.673,
                                "2009": 1,
                                "2010": 1,
                                "2011": 1,
                                "2012": 1,
                                "2013": 0.998,
                                "2014": 1,
                              },
                            }
          }
          compareCard1={undefined}
          compareCard2={undefined}
          displaySelected={undefined}
/>,
      ],
      "className": "card-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "cardInfo": Object {
            "location": "ADAMS COUNTY 14",
            "stats": Object {
              "2004": 0.228,
              "2005": 0.3,
              "2006": 0.293,
              "2007": 0.306,
              "2008": 0.673,
              "2009": 1,
              "2010": 1,
              "2011": 1,
              "2012": 1,
              "2013": 0.998,
              "2014": 1,
            },
          },
          "compareCard1": undefined,
          "compareCard2": undefined,
          "displaySelected": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "cardInfo": Object {
            "location": "ADAMS COUNTY 14",
            "stats": Object {
              "2004": 0.228,
              "2005": 0.3,
              "2006": 0.293,
              "2007": 0.306,
              "2008": 0.673,
              "2009": 1,
              "2010": 1,
              "2011": 1,
              "2012": 1,
              "2013": 0.998,
              "2014": 1,
            },
          },
          "compareCard1": undefined,
          "compareCard2": undefined,
          "displaySelected": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            cardInfo={
                        Object {
                                    "location": "ADAMS COUNTY 14",
                                    "stats": Object {
                                      "2004": 0.228,
                                      "2005": 0.3,
                                      "2006": 0.293,
                                      "2007": 0.306,
                                      "2008": 0.673,
                                      "2009": 1,
                                      "2010": 1,
                                      "2011": 1,
                                      "2012": 1,
                                      "2013": 0.998,
                                      "2014": 1,
                                    },
                                  }
            }
            compareCard1={undefined}
            compareCard2={undefined}
            displaySelected={undefined}
/>,
          <Card
            cardInfo={
                        Object {
                                    "location": "ADAMS COUNTY 14",
                                    "stats": Object {
                                      "2004": 0.228,
                                      "2005": 0.3,
                                      "2006": 0.293,
                                      "2007": 0.306,
                                      "2008": 0.673,
                                      "2009": 1,
                                      "2010": 1,
                                      "2011": 1,
                                      "2012": 1,
                                      "2013": 0.998,
                                      "2014": 1,
                                    },
                                  }
            }
            compareCard1={undefined}
            compareCard2={undefined}
            displaySelected={undefined}
/>,
        ],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "cardInfo": Object {
              "location": "ADAMS COUNTY 14",
              "stats": Object {
                "2004": 0.228,
                "2005": 0.3,
                "2006": 0.293,
                "2007": 0.306,
                "2008": 0.673,
                "2009": 1,
                "2010": 1,
                "2011": 1,
                "2012": 1,
                "2013": 0.998,
                "2014": 1,
              },
            },
            "compareCard1": undefined,
            "compareCard2": undefined,
            "displaySelected": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "cardInfo": Object {
              "location": "ADAMS COUNTY 14",
              "stats": Object {
                "2004": 0.228,
                "2005": 0.3,
                "2006": 0.293,
                "2007": 0.306,
                "2008": 0.673,
                "2009": 1,
                "2010": 1,
                "2011": 1,
                "2012": 1,
                "2013": 0.998,
                "2014": 1,
              },
            },
            "compareCard1": undefined,
            "compareCard2": undefined,
            "displaySelected": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
